### ISSUE TITLE
EXC_BAD_ACCESS on release executions

### ISSUE THREAD

**Issue by LightMan** [hollance/Forge#27]
I tried to run this project using the release version and they all crash at:

 `    mpscnn = MPSCNNFullyConnected(device: device,
                                  convolutionDescriptor: desc,
                                  kernelWeights: weights.pointer,
                                  biasTerms: biasTerms,
                                  flags: .none)
`

In createCompute of layers.swift

Only MNIST project, which does not use Forge, seems to work.

This means that an app using this library cannot be compiled to be sent to the app store, ad hoc or enterprise distribution.

**Comment by hollance** [comment]
What version of iOS are you trying this on? I think I know why this happens but didn't see it on my devices yet.

**Comment by LightMan** [comment]
Hi @hollance 
I am using an iPhone 6 64gb running iOS 11.1 (15B93)

**Comment by hollance** [comment]
This is actually an issue I ran into before and thought I had fixed but apparently not. With optimizations enabled, Swift deallocates the object containing the bias terms for the layer *before* the layer gets constructed. This seems like a compiler error since there are still (optional) objects with a reference to this object and so it gets deallocated too soon. It also doesn't happen with optimizations disabled. I'll see if I can find a workaround.
